src/java.base/share/classes/java/lang/Process.java

```diff
@@ -153,14 +183,251 @@ public Process() {}
      * input stream to be buffered.
      *
      * @return the input stream connected to the error output of
      *         the process
      */
     public abstract InputStream getErrorStream();
 
+    /**
+     * Returns a {@link BufferedReader BufferedReader} connected to the standard
+     * output of the process. The {@link Charset} for the native encoding is used
+     * to read characters, lines, or stream lines from standard output.
+     *
+     * <p>This method delegates to {@link #inputReader(Charset)} using the
+     * {@link Charset} named by the {@code native.encoding}
+     * system property or the {@link Charset#defaultCharset()} if the
+     * {@code native.encoding} is not supported.
+     *
+     * @return a {@link BufferedReader BufferedReader} using the
+     *          {@code native.encoding} if supported, otherwise, the
+     *          {@link Charset#defaultCharset()}
+     * @since 17
+     */
+    public final BufferedReader inputReader() {
+        return inputReader(CharsetHolder.nativeCharset());
+    }
+
+    /**
+     * Returns a {@link BufferedReader BufferedReader} connected to the
+     * standard output of this process using a Charset.
+     * The {@code BufferedReader} can be used to read characters, lines,
+     * or stream lines of the standard output.
+     *
+     * <p>Characters are read by an InputStreamReader that reads and decodes bytes
+     * from this process {@link #getInputStream()}. Bytes are decoded to characters
+     * using the {@code charset}; malformed-input and unmappable-character
+     * sequences are replaced with the charset's default replacement.
+     * The {@code BufferedReader} reads and buffers characters from the InputStreamReader.
+     *
+     * <p>If the standard output of the process has been redirected using
+     * {@link ProcessBuilder#redirectOutput(Redirect) ProcessBuilder.redirectOutput}
+     * then the {@code InputStreamReader} will be reading from a
+     * <a href="ProcessBuilder.html#redirect-output">null input stream</a>.
+     *
+     * <p>Otherwise, if the standard error of the process has been redirected using
+     * {@link ProcessBuilder#redirectErrorStream(boolean)
+     * ProcessBuilder.redirectErrorStream} then the input reader returned by
+     * this method will receive the merged standard output and the standard error
+     * of the process.
+     *
+     * @apiNote
+     * Using both {@link #getInputStream} and {@link #inputReader} has
+     * unpredictable behavior since the buffered reader reads ahead from the
+     * input stream.
+     *
+     * <p>When the process has terminated, and the standard input has not been redirected,
+     * the bytes available from the underlying stream is on a best effort basis and
+     * may be unpredictable.
+     *
+     * @implNote
+     * This is equivalent to:
+     * {@code
+     *     return new BufferedReader(new InputStreamReader(getInputStream(), charset));
+     * }
```

Comment:
Does not seem to be equivalent any longer. Also, should it describe the behavior that it rejects the different Charset after the first invocation?